backend-rust/src/graphql_api/baker.rs

```diff
@@ -472,27 +475,135 @@ struct BakerPool<'a> {
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, DelegationSummary>> {
+        let pool = get_pool(ctx)?;
+        let config = get_config(ctx)?;
+        let query = ConnectionQuery::<DescendingI64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.delegators_connection_limit,
+        )?;
+        let mut row_stream = sqlx::query_as!(
+            DelegationSummary,
+            "SELECT * FROM (
+                SELECT
+                    index,
+                    address as account_address,
+                    delegated_restake_earnings as restake_earnings,
+                    delegated_stake as staked_amount
+                FROM accounts
+                WHERE delegated_target_baker_id = $5 AND
+                    accounts.index > $2 AND accounts.index < $1
+                ORDER BY
+                    (CASE WHEN $4 THEN accounts.index END) ASC,
+                    (CASE WHEN NOT $4 THEN accounts.index END) DESC
+                LIMIT $3
+            ) AS delegators
+            ORDER BY delegators.index DESC",
+            i64::from(query.from),
+            i64::from(query.to),
+            query.limit,
+            query.is_last,
+            self.id
+        )
+        .fetch(pool);
+        let mut connection = connection::Connection::new(false, false);
+        while let Some(delegator) = row_stream.try_next().await? {
+            connection.edges.push(connection::Edge::new(delegator.index.to_string(), delegator));
+        }
+        if let Some(page_max_index) = connection.edges.first() {
+            if let Some(max_index) = sqlx::query_scalar!(
+                "SELECT MAX(index) FROM accounts WHERE delegated_target_baker_id = $1",
+                self.id
+            )
+            .fetch_one(pool)
+            .await?
+            {
+                connection.has_previous_page = max_index > page_max_index.node.index;
+            }
+        }
+        if let Some(edge) = connection.edges.last() {
+            connection.has_next_page = edge.node.index != 0;
+        }
+        Ok(connection)
+    }
+}
+
+struct DelegationSummary {
+    index:            i64,
+    account_address:  AccountAddress,
+    staked_amount:    i64,
+    restake_earnings: Option<bool>,
+}
+
+#[Object]
+impl DelegationSummary {
+    async fn account_address(&self) -> &AccountAddress { &self.account_address }
+
+    async fn staked_amount(&self) -> ApiResult<Amount> {
+        self.staked_amount.try_into().map_err(|_| {
+            ApiError::InternalError(
+                "Staked amount in database should be a valid UnsignedLong".to_string(),
```

Comment:
amount on accounts is a BitInt which is a signed long. So you can actually cause this error to trigger.

Reply:
A bit of a thought of issue, since not even enough CCD exists to cause this to occur.